model/map-jpa/src/main/resources/META-INF/authz/policy/jpa-authz-policy-changelog-1.xml

```diff
@@ -0,0 +1,126 @@
+
+        <createTable tableName="kc_authz_policy">
+            <column name="id" type="UUID">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="version" type="INTEGER" defaultValueNumeric="0">
+                <constraints nullable="false"/>
+            </column>
+            <column name="metadata" type="json"/>
+        </createTable>
+        <ext:addGeneratedColumn tableName="kc_authz_policy">
+            <ext:column name="entityversion" type="INTEGER" jsonColumn="metadata" jsonProperty="entityVersion"/>
+            <ext:column name="realmid" type="KC_KEY" jsonColumn="metadata" jsonProperty="fRealmId"/>
+            <ext:column name="resourceserverid" type="UUID" jsonColumn="metadata" jsonProperty="fResourceServerId"/>
+            <ext:column name="name" type="VARCHAR(255)" jsonColumn="metadata" jsonProperty="fName"/>
+            <ext:column name="owner" type="KC_KEY" jsonColumn="metadata" jsonProperty="fOwner"/>
+            <ext:column name="type" type="VARCHAR(255)" jsonColumn="metadata" jsonProperty="fType"/>
+        </ext:addGeneratedColumn>
+        <createIndex tableName="kc_authz_policy" indexName="authz_policy_entityVersion">
+            <column name="entityversion"/>
+        </createIndex>
+        <createIndex tableName="kc_authz_policy" indexName="authz_policy_realm_resource_server_name" unique="true">
+            <column name="realmid"/>
+            <column name="resourceserverid"/>
+            <column name="name"/>
+        </createIndex>
+        <createIndex tableName="kc_authz_policy" indexName="authz_policy_type">
+            <column name="type"/>
+        </createIndex>
+        <addForeignKeyConstraint constraintName="authz_policy_fk_root_fkey"
+                                 baseTableName="kc_authz_policy"
+                                 baseColumnNames="resourceserverid"
+                                 referencedTableName="kc_authz_resource_server"
+                                 referencedColumnNames="id"
+                                 onDelete="CASCADE"/>
+
+        <createTable tableName="kc_authz_policy_scope">
+            <column name="policy_id" type="UUID">
+                <constraints nullable="false" foreignKeyName="authz_policy_scope_policy_fk" references="kc_authz_policy(id)" deleteCascade="true"/>
+            </column>
+            <column name="scope_id" type="UUID">
+                <constraints nullable="false" foreignKeyName="authz_policy_scope_scope_fk" references="kc_authz_scope(id)" deleteCascade="true"/>
+            </column>
+        </createTable>
+        <createIndex tableName="kc_authz_policy_scope" indexName="authz_policy_scope_policy_id">
+            <column name="policy_id"/>
+        </createIndex>
+        <createIndex tableName="kc_authz_policy_scope" indexName="authz_policy_scope_scope_id">
+            <column name="scope_id"/>
+        </createIndex>
```

Comment:
AFAIK policy_id + scope_id combined should be unique in the table, as it is used as a `Set`. Please add a DB constraint for that, with the usually deferrable and initiallyDeferred as Hibernate might change the order of delete and insert.

This might be true for other `ElementCollection` tables as well.

Reply:
Added, thanks